test/integration/commands/excel2json/test_new_lists.py

```diff
@@ -0,0 +1,46 @@
+import json
+from pathlib import Path
+from typing import Any
+from typing import cast
+
+import pytest
+
+from dsp_tools.commands.excel2json.new_lists import new_excel2lists
+
+
+@pytest.fixture()
+def result_lists_correct() -> list[dict[str, Any]]:
+    all_lists, _ = new_excel2lists(Path("testdata/excel2json/new_excel2json_files/lists"))
+    all_lists = sorted(all_lists, key=lambda x: x["name"])
+    return all_lists
+
+
+@pytest.fixture()
+def expected_lists() -> list[dict[str, Any]]:
+    with open("testdata/excel2json/new-lists-output-expected.json", encoding="utf-8") as f:
+        expected = json.load(f)
+        return cast(list[dict[str, Any]], expected)
+
+
+def test_number_of_lists(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
+    assert len(result_lists_correct) == len(expected_lists)
+
+
+def test_identical_lists(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
+    assert result_lists_correct == expected_lists
+
+
+def test_first_list(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
+    assert result_lists_correct[0] == expected_lists[0]
+
+
+def test_second_list(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
+    assert result_lists_correct[1] == expected_lists[1]
+
+
+def test_thrid_list(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
+    assert result_lists_correct[2] == expected_lists[2]
```

Comment:
This testing strategy seems a bit unusual to me. I guess you want to know exactly where it fails. But there are some downsides:

 - you still have several failing tests, and it isn't clear that there is only 1 root cause for all these failures.
 - The order of the tests is not logical: If you want to know where exactly it fails, I would first test the length, then each element of the list, and only at the very end `assert result_lists_correct == expected_lists`
 - the price you pay is that the fixture is executed multiple times (not sure, but I think), and that you have many LOC which are redundant.

What about this:

```suggestion
def test_new_lists(expected_lists: list[dict[str, Any]], result_lists_correct: list[dict[str, Any]]) -> None:
    assert len(result_lists_correct) == len(expected_lists)
    assert result_lists_correct[0] == expected_lists[0]
    assert result_lists_correct[1] == expected_lists[1]
    assert result_lists_correct[2] == expected_lists[2]
    assert result_lists_correct == expected_lists
```

Then you see on what line it failed, so you know exactly where it failed.

Reply:
I prefer to have the assertions separate. Even if two tests fail because of one root cause. But We could also delete `result_lists_correct == expected_lists` because in this case there is nothing that is not tested.

Reply:
I changed it to an assertion against the number, this way we know that there aren't more lists that were added. So we can ensure that each individual list has been tested.